src/boot/AppEventHandlers.js

```diff
@@ -69,13 +69,30 @@ type Props = $ReadOnly<{|
   unreadCount: number,
 |}>;
 
+type OrientationLookup = {|
+  [expoKey: $Values<typeof ScreenOrientation.Orientation>]: OrientationT,
+|};
+
+const orientationLookup: OrientationLookup = {
+  [ScreenOrientation.Orientation.UNKNOWN]: 'PORTRAIT',
+  [ScreenOrientation.Orientation.PORTRAIT_UP]: 'PORTRAIT',
+  [ScreenOrientation.Orientation.PORTRAIT_DOWN]: 'PORTRAIT',
+  [ScreenOrientation.Orientation.LANDSCAPE_LEFT]: 'LANDSCAPE',
+  [ScreenOrientation.Orientation.LANDSCAPE_RIGHT]: 'LANDSCAPE',
+};
+
 class AppEventHandlers extends PureComponent<Props> {
   /** NetInfo disconnection callback. */
   netInfoDisconnectCallback: (() => void) | null = null;
 
-  handleOrientationChange = (orientation: OrientationT) => {
+  handleOrientationChange = (event: ScreenOrientation.OrientationChangeEvent) => {
     const { dispatch } = this.props;
-    dispatch(appOrientation(orientation));
+
+    const {
+      orientationInfo: { orientation },
+    } = event;
```

Comment:
The multi-line formatting is kind of a sign here of the fact that this syntax takes a bit  of effort for the reader to unpack :wink: 

The usual way we'd write this is:
```js
const { orientation } = event.orientationInfo;
```